### PR TITLE
kbld: init at 0.37.0

### PR DESCRIPTION
**Pull request by selmison**
###### Description of changes
kbld seamlessly incorporates image building and image pushing into your development and deployment workflows.

[Kbld](https://carvel.dev/kbld/) is part of the [carvel](https://carvel.dev/) collection of tooling with ytt, kapp and vendir which are already in nixpkgs.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/kbld/default.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+  vendorHash = null;
+
+  subPackages = [ "cmd/kbld" ];
+
+  ldflags = [
+    "-X github.com/vmware-tanzu/carvel-kbld/pkg/kbld/version.Version=${version}"
+  ];
+
+  nativeBuildInputs = [ installShellFiles ];
```
Comment (IndeedNotJames) [inline pkgs/development/tools/kbld/default.nix:22]:
```suggestion
```

Isn't used.

File: pkgs/development/tools/kbld/default.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
+{ lib, buildGoModule, fetchFromGitHub, installShellFiles, testers, kbld }:
```
Comment (IndeedNotJames) [inline pkgs/development/tools/kbld/default.nix:1]:
```suggestion
{ lib, buildGoModule, fetchFromGitHub, testers, kbld }:
```

File: pkgs/top-level/all-packages.nix
Comment (IndeedNotJames) [inline pkgs/top-level/all-packages.nix:1]: Please don't re-order existing attributes.
It's a lost cause at this point :^(

See [RFC 140](https://github.com/NixOS/rfcs/pull/140)

**Review by emilylange — CHANGES_REQUESTED** [review]
Looks pretty good so far.

Just two things:

File: pkgs/development/tools/kbld/default.nix [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  ];
+
+  nativeBuildInputs = [ installShellFiles ];
+
+  passthru.tests.version = testers.testVersion {
+    package = kbld;
+  };
+
+  meta = with lib; {
+    description = "kbld seamlessly incorporates image building and image pushing into your development and deployment workflows";
```
Comment (IndeedNotJames) [inline pkgs/development/tools/kbld/default.nix:29]:
```suggestion
    description = "Seamlessly incorporate image building and image pushing into your development and deployment workflows";
```

From <https://nixos.org/manual/nixpkgs/stable/#var-meta-description>:

> 
> A short (one-line) description of the package. This is shown by `nix-env -q --description` and also on the Nixpkgs release pages.
> 
> Don't include a period at the end. Don't include newline characters. Capitalise the first character. For brevity, don't repeat the name of package --- just describe what it does.
> 
> Wrong: `"libpng is a library that allows you to decode PNG images."`
> 
> Right: `"A library for decoding PNG images"`

**Comment by piegamesde** [comment]
Hi, it looks like you accidentally pushed a lot of commits, which accidentally pinged a bunch of people. Since they can't be unpinged, please open a new pull request with your changes.

https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md also contains instructions for how to properly rebase or change target branches without mass-pinging people.

**Pull request closed without merging**